### ISSUE TITLE
Some users on certain teams builds don't send messageBack messages correctly

### ISSUE THREAD

**Issue by clintwar** [MicrosoftDocs/msteams-docs#10902]
### Steps to reproduce

Send adaptive card with messageBack action with data of

msteams : {
text: "some text",
displayText: "some text",
value: null
}

### Expected behavior

The message sent to v1/agents.... endpoint should have text property set to "some text"

### Actual behavior

The message sent to v1/agents.... endpoint has text set to text:{}

### Error details

No error details, however the result is the bot to which the messageBack was received no longer works.

Seems to happen on teams client version 1415/24050922002

Cannot reproduce on client version 1415/2405030610

Happens when user is using desktop teams or teams in web browser.

**Comment by Meghana-MSFT** [comment]
Thank you for reporting this, we will check this and get back to you.

**Comment by Meghana-MSFT** [comment]
We checked the messageBack action with Adaptive Cards. It is working fine at our end. To include a messageBack action with an Adaptive Card, type	needs to be set to messageBack in msteams object - https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-actions?tabs=json#adaptive-cards-with-messageback-action

Could you please share your entire JSON?

**Comment by clintwar** [comment]
{
  "type": "AdaptiveCard",
  "version": "1.0",
  "body": [
    {
      "type": "TextBlock",
      "size": "medium",
      "weight": "bolder",
      "text": "[TEST Sub Header Goes Here]",
      "wrap": true
    },
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": "auto",
          "items": [
            {
              "type": "Image",
              "size": "small",
              "style": "person",
              "url": "https://testimagepage/img.png"
            }
          ]
        },
        {
          "type": "Column",
          "width": "auto",
          "items": [
            {
              "type": "TextBlock",
              "size": "medium",
              "weight": "bolder",
              "text": "[TEST Title Goes Here]",
              "wrap": true
            },
            {
              "type": "TextBlock",
              "size": "small",
              "weight": "bolder",
              "color": "attention",
              "text": "",
              "wrap": true,
              "spacing": "small"
            }
          ]
        }
      ]
    },
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": "32px",
          "items": [
            {
              "type": "TextBlock",
              "size": "large",
              "color": "accent",
              "text": "1",
              "horizontalAlignment": "center",
              "wrap": true
            }
          ]
        },
        {
          "type": "Column",
          "width": "auto",
          "items": [
            {
              "type": "TextBlock",
              "size": "medium",
              "text": "Item One",
              "wrap": true
            },
            {
              "type": "TextBlock",
              "size": "small",
              "isSubtle": true,
              "text": "This Is Items One Description",
              "wrap": true,
              "spacing": "small"
            }
          ],
          "spacing": "medium"
        }
      ],
      "selectAction": {
        "type": "Action.Submit",
        "data": {
          "msteams": {
            "displayText": "Item One",
            "text": "1",
            "type": "messageBack"
          }
        }
      },
      "spacing": "large",
      "separator": true
    }
  ]
}

Again this works for people using certain client versions of teams, and it does not work for people using other client versions of teams.

**Comment by Meghana-MSFT** [comment]
We are on the same version where it is working. We checked in following versions 1415/24050307610, 1415/24050307611.
We will try in other versions and get back to you.

**Comment by clintwar** [comment]
Please try 1415/24050922002. The end user that could reproduce was on client version 1415/24050922002. I mentioned this above in my original post.

**Comment by Meghana-MSFT** [comment]
Yes, we are checking if we can find the above versions. We will check and get back to you.

**Comment by Meghana-MSFT** [comment]
@clintwar - Could you please confirm if the users are still facing this issue? Are there any other versions where it is reproducible? We checked in following version and cannot reproduce the issue 1415/24050307614.

**Comment by clintwar** [comment]
end user confirmed no repro on their current version. Their version has increased so I am going to assume that someone else found the issue and fixed it.

Version
You have Microsoft Teams version 24152.407.2925.6763.
You've got the latest version. It was last checked on 6/10/2024.
The client version is 49/24053101407.

**Comment by Meghana-MSFT** [comment]
Thank you for confirming that the issue is no longer reproducible. Please feel free to reach out to us if you face this issue again.
